imblearn/metrics/_classification.py

```diff
@@ -997,3 +998,57 @@ class 2       1.00      0.67      1.00      0.80      0.82      0.64\
     if output_dict:
         return report_dict
     return report
+
+
+@_deprecate_positional_args
+def macro_averaged_mean_absolute_error(y_true, y_pred):
+    """Compute Macro-Averaged Mean Absolute Error (MA-MAE)
+    for imbalanced ordinal classification.
+
+    This function computes each MAE for each class and average them,
+    giving an equal weight to each class.
+
+    Read more in the :ref:`User Guide <macro_averaged_mean_absolute_error>`.
+
+    Parameters
+    ----------
+    y_true : 1d array-like, or label indicator array / sparse matrix
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like, or label indicator array / sparse matrix
+        Estimated targets as returned by a classifier.
+
+    Returns
+    -------
+    loss : float or ndarray of floats
+        Macro-Averaged MAE output is non-negative floating point.
+        The best value is 0.0.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import mean_absolute_error
+    >>> from imblearn.metrics import macro_averaged_mean_absolute_error
+    >>> y_true_balanced = [1, 1, 2, 2]
+    >>> y_true_imbalanced = [1, 2, 2, 2]
+    >>> y_pred = [1, 2, 1, 2]
+    >>> mean_absolute_error(y_true_balanced, y_pred)
+       0.5
+    >>> mean_absolute_error(y_true_imbalanced, y_pred)
+       0.25
+    >>> macro_averaged_mean_absolute_error(y_true_balanced, y_pred)
+       0.5
+    >>> macro_averaged_mean_absolute_error(y_true_imbalanced, y_pred)
+       0.16666666666666666
+
+    """
+    all_mae = []
+    y_true = np.array(y_true)
+    y_pred = np.array(y_pred)
+    for class_to_predict in np.unique(y_true):
```

Comment:
I think that we should introduce a `label` to be able to either give class that are not present in `y_true` or select a subset of labels as in precision-recall: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_fscore_support.html#sklearn.metrics.precision_recall_fscore_support

Would it make sense?